Sources/SpeziOnboarding/OnboardingConsentView.swift

```diff
@@ -0,0 +1,158 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import SwiftUI
+
+
+/// The ``OnboardingConsentView`` provides a convenient onboarding view for the display of markdown-based documents that can be
+/// signed using a family and given name and a hand drawn signature. Furthermore, the view includes an export functionality, enabling users
+/// to share and store the signed consent form.
+/// The exported consent form is automatically stored in the Spezi `Standard`, requiring the `Standard` to conform to the ``OnboardingConstraint``.
+///
+/// The ``OnboardingConsentView`` builds on top of the SpeziOnboarding ``ConsentDocument`` 
+/// by providing a more developer-friendly, convenient API with additional functionalities like the share consent option.
+///
+/// ```swift
+/// OnboardingConsentView(
+///     markdown: {
+///         Data("This is a *markdown* **example**".utf8)
+///     },
+///     action: {
+///         // The action that should be performed once the user has provided their consent.
+///     },
+///     exportConfiguration: .init(paperSize: .usLetter)   // Configure the properties of the exported consent form
+/// )
+/// ```
+public struct OnboardingConsentView: View {
+    private let markdown: (() async -> Data)
+    private let action: (() async -> Void)
+    private let exportConfiguration: ConsentDocument.ExportConfiguration
+    
+    @EnvironmentObject private var onboardingDataSource: OnboardingDataSource
+    @State private var viewState: ConsentViewState = .base(.idle)
+    @State private var willShowShareSheet = false
+    @State private var showShareSheet = false
+    
+    
+    public var body: some View {
+        ScrollViewReader { proxy in // swiftlint:disable:this closure_body_length
+            OnboardingView(
+                titleView: {
+                    OnboardingTitleView(
+                        title: LocalizedStringResource("CONSENT_VIEW_TITLE", bundle: .atURL(from: .module))
+                    )
+                },
```

Comment:
I think it would make sense to move this element out of the Onboarding Consent View, one might want to adjust the title or change out the title view to me something completely different and not even set it at all.

I would suggest to remove this title view for now and if an App e.g. wants to display it it can use the Title View itself or provide some more custom view.

Alternative would be to make the title configurable and optional and not add a title view if nil is passed in.

Reply:
Yes, that seems like a good idea, however, I went for a default value of `LocalizedStringResource("CONSENT_TITLE", bundle: .atURL(from: .module))` instead of `nil`. 
As the `OnboardingConsentView` is a full-fledged onboarding view (in comparison to the more "bare bone" `ConsentDocument`), I think this is the better approach here. Developers can still pass `nil` to it, therefore disabeling the title completly.

Reply:
Sounds good, thank you! 🚀